torch_ttnn/passes/lowering/add_data_move_pass.py

```diff
@@ -315,252 +289,231 @@ def is_target_a_user_of_curr_node(curr_node, target):
-    new_nodes = list()
-    with g.inserting_before(dst_node):
-        kwargs = {"layout": TtnnTileLayout(), "device": device}
-        new_nodes.append(g.call_function(ttnn.from_torch, (src_node,), kwargs))
-
-    insert_node_between_kwarg(src_node, key, dst_node, new_nodes)
-    return new_nodes[-1]
-
-
-def try_add_data_move_in(src_node, dst_idx, dst_node, device) -> torch.fx.node.Node:
-    if not should_add_data_move_in(src_node, dst_node):
-        return None
-
-    g = dst_node.graph
-    new_nodes = list()
-    with g.inserting_before(dst_node):
-        kwargs = {"layout": TtnnTileLayout(), "device": device}
-
-        if is_target_a_user_of_curr_node(dst_node, ttnn.embedding) and dst_idx == 0:
-            kwargs["dtype"] = TtnnUint32()
-        else:
-            kwargs["dtype"] = TtnnBfloat16()
-
-        new_nodes.append(g.call_function(ttnn.from_torch, (src_node,), kwargs))
-
-    insert_node_between(src_node, dst_idx, dst_node, new_nodes)
-    return new_nodes[-1]
-
-
-def try_add_layout_change_before_node(src_node, dst_idx, dst_node, device) -> torch.fx.node.Node:
-    # Consider dst_node is ttnn.repeat, and src_node are any tt nodes that ttnn.repeat uses
-    if isinstance(src_node, (int, float, list, tuple)) or not isinstance(src_node, torch.fx.node.Node):
-        return None
-    if not is_function_call(dst_node):
-        return None
-
-    need_from_device = False
-    need_to_layout = False
-    need_to_device = False
-    # TODO(#372): #322 will enable tile layout for more layout change ops
-    if dst_node.target in TTNN_LAYOUT_CHANGE_OPS and dst_idx == 0 and is_tt(src_node):
-        need_from_device = True
-        need_to_layout = True
-
-    if dst_node.target in [ttnn.embedding, ttnn.zeros_like, target_wrappers.repeat]:
-        # TODO: Only uint32 needs to to_layout on host
-        need_from_device = True
-        need_to_layout = True
-        need_to_device = True
-
-    if not any((need_from_device, need_to_layout, need_to_device)):
-        return None
-
-    g = dst_node.graph
-    with g.inserting_before(dst_node):
-        new_nodes = [src_node]
-        if need_from_device:
-            new_nodes.append(g.call_function(ttnn.from_device, (new_nodes[-1],)))
-        if need_to_layout:
-            new_nodes.append(g.call_function(ttnn.to_layout, (new_nodes[-1], TtnnRowMajorLayout())))
-        if need_to_device:
-            new_nodes.append(g.call_function(ttnn.to_device, (new_nodes[-1],), {"device": device}))
-        new_nodes = new_nodes[1:]
-
-    insert_node_between(src_node, dst_idx, dst_node, new_nodes)
-    return new_nodes[-1]
-
-
-def try_add_layout_change_after_node(src_node, dst_idx, dst_node, device) -> torch.fx.node.Node:
-    # Consider src_node is ttnn.repeat, and dst_node should be any tt_compute node that uses ttnn.repeat
-    if not is_function_call(src_node):
-        return None
-    if (
-        src_node.target
-        not in TTNN_LAYOUT_CHANGE_OPS.union(
+class NodeInputAligner:
```

Comment:
move to a different file?

Reply:
This is #532 part, I'll merge that PR first adn then this PR